streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -150,8 +150,11 @@ public InternalStreamsBuilder(final InternalTopologyBuilder internalTopologyBuil
         Objects.requireNonNull(materialized, "materialized can't be null");
         // explicitly disable logging for global stores
         materialized.withLoggingDisabled();
-        final String sourceName = newProcessorName(KTableImpl.SOURCE_NAME);
-        final String processorName = newProcessorName(KTableImpl.SOURCE_NAME);
+        final String sourceName = new NamedInternal(consumed.name())
+                .orElseGenerateWithPrefix(this, KStreamImpl.SOURCE_NAME);
+        final String processorName = new NamedInternal(consumed.name())
+                .suffixWithOrElseGet("-table-source", this, KTableImpl.SOURCE_NAME);
```

Comment:
shouldn't the `table-source` suffix go with the `sourceName`?

Reply:
@bbejeck, Yes, that seems more logical. I made the modification for the global table too. Also, I think the suffix "-source" is sufficient.